File: PrimeFactors/PrimeFactors.ts
export class PrimeFactors {
  generate(n: number): number[] {
    let primes = []

    for (let candidate = 2; n > 1; candidate++) {
      for (; n % candidate === 0; n /= candidate) {
        primes.push(candidate)
      }
    }

    return primes
  }
}

File: PrimeFactors/13062022/prime_factors.go
package main

func PrimeFactors(number int) []int {
	factors := make([]int, 0)

	for test := 2; test <= number; test++ {
		for ; number%test == 0; number /= test {
			factors = append(factors, test)
		}
	}

	return factors
}
